modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/JsonProcessorTests.java

```diff
@@ -33,7 +36,7 @@ public void testExecute() throws Exception {
         String processorTag = randomAlphaOfLength(3);
         String randomField = randomAlphaOfLength(3);
         String randomTargetField = randomAlphaOfLength(2);
-        JsonProcessor jsonProcessor = new JsonProcessor(processorTag, null, randomField, randomTargetField, false, REPLACE, false);
+        JsonProcessor jsonProcessor = new JsonProcessor(processorTag, null, randomField, randomTargetField, false, REPLACE, false, true);
```

Comment:
These lines in this test file make me wonder if we should add a constructor to `JsonProcessor` which has the default `strict` behavior, and then have the other constructor for allowing the caller to set it. It would at the very least reduce the changes on this file.

Reply:
Yeah I can do that -- I went back and forth over whether it would be simpler to have another constructor or another required argument to the single constructor.